server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamAction.java

```diff
@@ -64,22 +66,32 @@ public static Type fromValue(byte value) {
     private final Type type;
     private String dataStream;
     private String index;
+    private boolean failureStore = false;
```

Comment:
I don't really foresee that kind of change happening - DataStreamAction defines an operation to be done for a data stream and an index. I can't think of any operations that we'd likely add here that 1) wont want to be done exclusively on either the backing indices or the failure store or 2) wouldn't be better off as part of a different configuration API. Assuming we did come up with something that we might want to do for both backing indices and failure store indices, that could be specified with multiple actions in the request which are applied at the same time.

I wanted to avoid using `FailureStoreOptions` here also because it's more of a component to the `IndicesOptions` and I didn't want to leak it overly much to different parts of the code. This API uses `IndicesOptions`, sure, but it does so only to resolve the specific index names provided to the actions. It doesn't support wildcards or the like, so the `IndicesOptions` were left alone, and I used a boolean here since we don't allow data streams to be specified in the `index` field. The data stream provided to the `data stream` field is not resolved to backing indices/failure store indices in a resolver object, so I additionally didn't use `FailureStoreOptions` there.